src/transformers/models/vits/modeling_vits.py

```diff
@@ -1435,7 +1435,9 @@ def forward(
         if self.config.num_speakers > 1 and speaker_id is not None:
             if not 0 <= speaker_id < self.config.num_speakers:
                 raise ValueError(f"Set `speaker_id` in the range 0-{self.config.num_speakers - 1}.")
-            speaker_embeddings = self.embed_speaker(torch.tensor([speaker_id])).unsqueeze(-1)
+            if isinstance(speaker_id, int):
+                speaker_id = torch.tensor([speaker_id], device=self.embed_speaker.weight.device)
```

Comment:
You suggested that `self.device` was clearer here! Happy to go with that :)

```suggestion
                speaker_id = torch.tensor([speaker_id], device=self.device)
```